imagekit-sdk/src/test/java/io/imagekit/sdk/ImageKitTest.java

```diff
@@ -2,629 +2,1890 @@
+				+ "Content-Disposition: form-data; name=\"folder\"\r\n" + "Content-Length: 5\r\n" + "\r\n" + "demo1\r\n"
+				+ "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"customCoordinates\"\r\n" + "Content-Length: 11\r\n" + "\r\n"
+				+ "10,10,20,20\r\n" + "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"responseFields\"\r\n" + "Content-Length: 32\r\n" + "\r\n"
+				+ "thumbnail,tags,customCoordinates\r\n" + "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"overwriteFile\"\r\n" + "Content-Length: 4\r\n" + "\r\n"
+				+ "true\r\n" + "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"overwriteAITags\"\r\n" + "Content-Length: 4\r\n" + "\r\n"
+				+ "true\r\n" + "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"overwriteTags\"\r\n" + "Content-Length: 4\r\n" + "\r\n"
+				+ "true\r\n" + "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"overwriteCustomMetadata\"\r\n" + "Content-Length: 4\r\n"
+				+ "\r\n" + "true\r\n" + "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"extensions\"\r\n" + "Content-Length: 114\r\n" + "\r\n"
+				+ "[{\"name\":\"remove-bg\",\"options\":{\"add_shadow\":true}},{\"name\":\"google-auto-tagging\",\"minConfidence\":10,\"maxTags\":5}]\r\n"
+				+ "--randomBoundary-------------------\r\n" + "Content-Disposition: form-data; name=\"webhookUrl\"\r\n"
+				+ "Content-Length: 57\r\n" + "\r\n" + "https://webhook.site/c78d617f-33bc-40d9-9e61-608999721e2e\r\n"
+				+ "--randomBoundary-------------------\r\n"
+				+ "Content-Disposition: form-data; name=\"customMetadata\"\r\n" + "Content-Length: 12\r\n" + "\r\n"
+				+ "{\"test1\":10}\r\n" + "--randomBoundary---------------------";
+		assertEquals(json, request.getBody().readUtf8().trim());
+		assertEquals("POST /api/v1/files/upload HTTP/1.1", request.getRequestLine());
+		assertEquals(RestClient.UPLOAD_BASE_URL.concat("api/v1/files/upload"), request.getRequestUrl().toString());
+	}
+
+	@Test(expected = NotFoundException.class)
+	public void imageKit_updateDetails_expected_404()
+			throws IOException, InterruptedException, BadRequestException, InternalServerException, UnknownException,
+			ForbiddenException, TooManyRequestsException, UnauthorizedException {
+
+		FileUpdateRequest fileUpdateRequest = new FileUpdateRequest("62a9c3ccd875ec6fd658c804");
+		MockWebServer server = new MockWebServer();
+		String responseJson = "{\n" + "    \"message\": \"The requested file does not exist.\",\n"
+				+ "    \"help\": \"For support kindly contact us at support@imagekit.io .\"\n" + "}";
+		server.enqueue(new MockResponse().setResponseCode(404).setBody(responseJson));
+		server.start();
+		RestClient.API_BASE_URL = server.url("/").toString();
+		SUT.updateFileDetail(fileUpdateRequest);
+		server.takeRequest();
+	}
+
```

Comment:
It seems that List file API test cases are missing. Also take care of using passing `tags` as an array and SDK converting it to comma separated values properly. Also same for boolean fields conversion.